stock_picking_actual_date/models/stock_scrap.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockScrap(models.Model):
+    _inherit = "stock.scrap"
+
+    actual_date = fields.Date(tracking=True)
+    is_editable_actual_date = fields.Boolean(
+        compute="_compute_is_editable_actual_date", string="Is Editable"
+    )
+
+    def _compute_is_editable_actual_date(self):
+        for rec in self:
+            rec.is_editable_actual_date = False
+            if rec.state != "done" or self.env.user.has_group(
+                "stock.group_stock_manager"
+            ):
+                rec.is_editable_actual_date = True
+
+    def _prepare_move_values(self):
+        vals = super()._prepare_move_values()
+        if self.actual_date:
+            vals["actual_date"] = self.actual_date
+        return vals
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "actual_date" in vals:
+            for rec in self:
+                if rec.state != "done":
+                    continue
+                account_moves = rec.move_id.account_move_ids
+                if not account_moves:
+                    continue
+                account_moves.button_draft()
+                for am in account_moves:
+                    am = am.with_context(skip_date_sequence_check=True)
+                    am.date = am.stock_move_id.actual_date
+                    if not am._sequence_matches_date():
+                        am.name = False
+                account_moves.action_post()
```

Comment:
Let's not repeat the same logic in different places. Please create a method in account.move to handle this.